modules/apps/headless/headless-admin-workflow/headless-admin-workflow-impl/src/main/java/com/liferay/headless/admin/workflow/internal/dto/v1_0/util/NodeUtil.java

```diff
@@ -15,25 +15,43 @@
 package com.liferay.headless.admin.workflow.internal.dto.v1_0.util;
 
 import com.liferay.headless.admin.workflow.dto.v1_0.Node;
+import com.liferay.headless.admin.workflow.internal.resource.v1_0.WorkflowDefinitionResourceImpl;
 import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.util.MapUtil;
+import com.liferay.portal.kernel.util.ResourceBundleUtil;
 import com.liferay.portal.kernel.workflow.WorkflowNode;
 
-import java.util.ResourceBundle;
+import java.util.Locale;
+import java.util.Map;
 
 /**
  * @author Feliphe Marinho
  */
 public class NodeUtil {
 
 	public static Node toNode(
-		Language language, String name, ResourceBundle resourceBundle,
-		WorkflowNode.Type type) {
+		Language language, Locale locale, WorkflowNode workflowNode) {
 
 		Node node = new Node();
 
-		node.setLabel(language.get(resourceBundle, name));
-		node.setName(name);
-		node.setType(Node.Type.create(type.name()));
+		if (MapUtil.isEmpty(workflowNode.getLabelMap())) {
+			node.setLabel(
```

Comment:
Hey @MarinhoFeliphe, please use the set methods that accept an UnsafeSupplier instead. Thx